docs/source/compression/compression_config_list.rst

```diff
@@ -1,11 +1,190 @@
+For example, if the sparse ratio is 0.8, and the pruning target is a Linear module weight, 80% weight value will be masked after pruning.
+
+max_sparse_ratio
+^^^^^^^^^^^^^^^^
+
+A float number between 0. ~ 1., for each pruning target, the sparse ratio after pruning will not be larger than this number,
+that means at most masked ``max_sparse_ratio`` pruning target value.
+
+min_sparse_ratio
+^^^^^^^^^^^^^^^^
+
+A float number between 0. ~ 1., for each pruning target, the sparse ratio after pruning will not be lower than this number,
+that means at least masked ``min_sparse_ratio`` pruning target value.
+
+sparse_threshold
+^^^^^^^^^^^^^^^^
+
+A float number, different from the ``sparse_ratio`` configure a specific sparsity, ``sparse_threshold`` usually used in some adaptive sparse cases.
+The value range is different for different pruning algorithms, please reference the pruner document to see how to configure it.
+In general, the higher the threshold, the higher the final sparsity.
+
+global_group_id
+^^^^^^^^^^^^^^^
+
+All pruning targets that have same ``global_group_id`` will be treat as a whole, and the ``sparse_ratio`` will be distributed across pruning targets.
+That means each pruning target might have different sparse ratio after pruning, but the group sparse ratio will be the configured ``sparse_ratio``.
+
+Note that the ``sparse_ratio`` in the same global group should be the same.
+
+dependency_group_id
+^^^^^^^^^^^^^^^^^^^
+
+All pruning targets that have same ``dependency_group_id`` will be treat as a whole, and the positions the targets' pruned will be the same.
+For example, layer A and layer B have same ``dependency_group_id``, and they want to be pruned output channels, then A and B will be pruned the same channel indexes.
+
+Note that the ``sparse_ratio`` in the same dependency group should be the same, and the positions can be pruned should be same,
+for example, pruning targets should have same output channel number when pruning output channel.
+
+This key usually be used on modules with add operation, i.e., skip connection.
+
+granularity
+^^^^^^^^^^^
+
+Used to control the granularity of the generated masked.
+
+``default``, ``in_channel``, ``out_channel`` and list of integer are supported:
+
+- default: The pruner will auto determine using which kind of granularity, usually consistent with the paper.
+- in_channel: The pruner will do pruning on the weight parameters 1 dimension.
+- out_channel: The pruner will do pruning on the weight parameters 0 dimension.
+- list of integer: Block sparse will be applied. For example, ``[4, 4]`` will apply 4x4 block sparse on the last two dimensions of the weight parameters.
+
+Note that ``in_channel`` or ``out_channel`` is not supported for input/output targets.
+``torch.nn.Embedding`` is special, it's output dimension on weight is 1, so if want to pruning Embedding output channel, please set ``in_channel`` for its granularity.
```

Comment:
would be better to add simple configuration examples here

Reply:
add

```
The following is an example for output channel pruning:

.. code-block:: python

    config = {
        'op_types': ['Conv2d'],
        'sparse_ratio': 0.5,
        'granularity': 'out_channel' # same as [1, -1, -1, -1]
    }
```